packages/@ember/-internals/container/lib/registry.ts

```diff
@@ -1,9 +1,10 @@
-import { Factory } from '@ember/-internals/owner';
+import type { Factory } from '@ember/-internals/owner';
 import { dictionary, intern } from '@ember/-internals/utils';
 import { assert, deprecate } from '@ember/debug';
-import { set } from '@ember/object';
+import type { set } from '@ember/object';
 import { DEBUG } from '@glimmer/env';
-import Container, { ContainerOptions, LazyInjection } from './container';
+import type { ContainerOptions, LazyInjection } from './container';
+import Container from './container';
```

Comment:
This is annoying: the rule doesn't seem to understand that these can be intermixed freely now. 🤔 

([playground](https://www.typescriptlang.org/play?#code/PTAEAEBcGcFoEsDmA7A9gJwKalrUArAV2klAAtMtRJVQBbAQwGttILQAjTAG1QHdQfDEwBQ8OgAcMpAORQAnhMzRgmOl3QB9Tag75MAY0gyA3CLGTpoAKLrKAeT2HIAGlABvaouwATTADMGQm5SBmgbewAVb1AAX1B-dFQ6UDk1DWBdfSNTCyl0Uk9MAA8lI2ile384hKSUmRKyyFhIb1yRRucKzCqAHlsNR2zIAD4ACgBKADoaawBHQgZubr7rKO9xibMgA))